src/dex/verified/pools/primary/primaryPool.ts

```diff
@@ -1,54 +1,321 @@
-//     poolState: PoolState,
-//     tokenIn: string,
-//     tokenOut: string,
-//   ): PoolPairData {
-//     let indexIn = 0;
-//     let indexOut = 0;
-//     let bptIndex = 0;
-//     const balances: bigint[] = [];
-//     const tokens = poolState.orderedTokens.map((tokenAddress, i) => {
-//       const t = pool.tokensMap[tokenAddress.toLowerCase()];
-//       if (t.address.toLowerCase() === tokenIn.toLowerCase()) indexIn = i;
-//       if (t.address.toLowerCase() === tokenOut.toLowerCase()) indexOut = i;
-//       if (t.address.toLowerCase() === pool.address.toLowerCase()) bptIndex = i;
-//       balances.push(poolState.tokens[t.address.toLowerCase()].balance);
-//       return t.address;
-//     });
-
-//     const poolPairData: PoolPairData = {
-//         tokens,
-//         balances,
-//         indexIn,
-//         indexOut,
-//         bptIndex,
-//         swapFee: poolState.swapFee,
-//         minOrderSize: poolState.minimumOrderSize,
-
-//     };
-//     return poolPairData;
-//   }
-
-// }
+import { Interface } from '@ethersproject/abi';
+import {
+  OrdersState,
+  PoolPairData,
+  PoolState,
+  SubgraphPoolBase,
+  TokenState,
+  VerifiedPoolTypes,
+  callData,
+} from '../../types';
+import { MathSol } from '../primary/primaryPoolMath';
+import PRIMARYISSUEPOOL from '../../../../abi/verified/PrimaryIssuePool.json';
+import { decodeThrowError } from '../../utils';
+import { SwapSide } from '@paraswap/core';
+
+//Todo: explain the math better(after testing secondary)
+
+export class PrimaryIssuePool {
+  vaultAddress: string;
+  vaultInterface: Interface;
+  poolInterface: Interface;
+
+  constructor(vaultAddress: string, vaultInterface: Interface) {
+    this.vaultAddress = vaultAddress;
+    this.vaultInterface = vaultInterface;
+    this.poolInterface = new Interface(PRIMARYISSUEPOOL.abi);
+  }
+
+  //Helper function to parse both primary and secondary issue pools data into params for onSell and onBuy functions.
+  parsePoolPairData(
+    pool: SubgraphPoolBase,
+    poolState: PoolState,
+    tokenIn: string,
+    tokenOut: string,
+  ): PoolPairData {
+    let indexIn = 0;
+    let indexOut = 0;
+    let bptIndex = 0;
+    let balances: bigint[] = [];
+    let decimals: number[] = [];
+    let scalingFactors: bigint[] = [];
+    const tokens = poolState.orderedTokens.map((tokenAddress, i) => {
+      const t = pool.tokensMap[tokenAddress.toLowerCase()];
+      if (t.address.toLowerCase() === tokenIn.toLowerCase()) indexIn = i;
+      if (t.address.toLowerCase() === tokenOut.toLowerCase()) indexOut = i;
+      if (t.address.toLowerCase() === pool.address.toLowerCase()) bptIndex = i;
+      balances.push(poolState.tokens[t.address.toLowerCase()].balance);
+      const _decimal = pool.tokens[i].decimals;
+      decimals.push(_decimal);
+      scalingFactors.push(BigInt(10 ** (18 - _decimal)));
+      return t.address;
+    });
+    const orders = pool.orders;
+    const secondaryTrades = pool.secondaryTrades;
```

Comment:
@Mohzcrea8me secondaryTrades in primary pool ? I think you need to clean up the code and comments

Reply:
Persepools is generally I will clean it up and optimize it for each pools